### PR TITLE
warehouse_grant: update warehouse grants in-place instead of recreating

### PR DESCRIPTION
**Pull request by igungor**
Related to #469
Updates #137

### CODE REVIEW

**Review by alldoami — CHANGES_REQUESTED** [review]
Could you also write a test for Update here? Same goes for the other resources you opened a PR for 😄 thank you!

**Comment by igungor** [comment]
> Could you also write a test for Update here? Same goes for the other resources you opened a PR for 😄 thank you!

Could you point me to any `update grant` test or give a hint to how to write update grant test, I'd gladly contribute tests too.

**Comment by alldoami** [comment]
@igungor here 😄  https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/main/pkg/resources/warehouse_grant_test.go

**Comment by igungor** [comment]
> @igungor here 😄 https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/main/pkg/resources/warehouse_grant_test.go

I know where to put the tests, I was expecting a sample resource update-grant test :)

**Comment by alldoami** [comment]
@igungor ah sorry, we don't have any example tests right now, would you need help writing them?

File: pkg/resources/warehouse_grant.go [hunk 1]
```diff
@@ -127,3 +127,48 @@ func DeleteWarehouseGrant(d *schema.ResourceData, meta interface{}) error {
...
+	builder := snowflake.WarehouseGrant(grantID.ResourceName)
+
+	// first revoke
+	if err := deleteGenericGrantRolesAndShares(
+		meta,
+		builder,
+		grantID.Privilege,
+		rolesToRevoke,
+		nil,
+	); err != nil {
```
Comment (alldoami) [inline pkg/resources/warehouse_grant.go:156]: NIT: To follow convention in this file, could you make this `if` separate from the `err` declaration? Same as the ones below.

**Review by alldoami — APPROVED** [review]

**Pull request merged**